model/shared/src/main/scala/explore/model/ObservingModeSummary.scala

```diff
@@ -0,0 +1,143 @@
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats.Eq
+import cats.derived.*
+import cats.syntax.option.*
+import clue.data.syntax.*
+import lucuma.core.enums.GmosAmpReadMode
+import lucuma.core.enums.GmosNorthFilter
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosNorthGrating
+import lucuma.core.enums.GmosRoi
+import lucuma.core.enums.GmosSouthFilter
+import lucuma.core.enums.GmosSouthFpu
+import lucuma.core.enums.GmosSouthGrating
+import lucuma.core.util.Display
+import lucuma.schemas.ObservationDB.Types.GmosNorthLongSlitInput
+import lucuma.schemas.ObservationDB.Types.GmosSouthLongSlitInput
+import lucuma.schemas.ObservationDB.Types.ObservingModeInput
+import lucuma.schemas.model.CentralWavelength
+import lucuma.schemas.model.ObservingMode
+import lucuma.schemas.odb.input.*
+
+// Observing mode with explicit values merged over defaults. Used for grouping observations by configuration.
+enum ObservingModeSummary derives Eq:
+  case GmosNorthLongSlit(
+    grating:           GmosNorthGrating,
+    filter:            Option[GmosNorthFilter],
+    fpu:               GmosNorthFpu,
+    centralWavelength: CentralWavelength,
+    ampReadMode:       GmosAmpReadMode,
+    roi:               GmosRoi
+  ) extends ObservingModeSummary
+  case GmosSouthLongSlit(
+    grating:           GmosSouthGrating,
+    filter:            Option[GmosSouthFilter],
+    fpu:               GmosSouthFpu,
+    centralWavelength: CentralWavelength,
+    ampReadMode:       GmosAmpReadMode,
+    roi:               GmosRoi
+  ) extends ObservingModeSummary
+
+  def toInput: ObservingModeInput = this match
+    case GmosNorthLongSlit(grating, filter, fpu, centralWavelength, ampReadMode, roi) =>
+      ObservingModeInput(
+        gmosNorthLongSlit = GmosNorthLongSlitInput(
+          grating = grating.assign,
+          filter = filter.orUnassign,
+          fpu = fpu.assign,
+          centralWavelength = centralWavelength.value.toInput.assign,
+          explicitAmpReadMode = ampReadMode.assign,
+          explicitRoi = roi.assign
+        ).assign
+      )
+    case GmosSouthLongSlit(grating, filter, fpu, centralWavelength, ampReadMode, roi) =>
+      ObservingModeInput(
+        gmosSouthLongSlit = GmosSouthLongSlitInput(
+          grating = grating.assign,
+          filter = filter.orUnassign,
+          fpu = fpu.assign,
+          centralWavelength = centralWavelength.value.toInput.assign,
+          explicitAmpReadMode = ampReadMode.assign,
+          explicitRoi = roi.assign
+        ).assign
+      )
+
+object ObservingModeSummary:
+  def fromObservingMode(observingMode: ObservingMode): ObservingModeSummary =
+    observingMode match
+      case ObservingMode.GmosNorthLongSlit(
+            _,
+            grating,
+            _,
+            filter,
+            _,
+            fpu,
+            _,
+            centralWavelength,
```

Comment:
It seems there maybe plenty of unused params

Reply:
Yes, the story requests:

```
The components that should be included are:
Grating, Filter, FPU, Central Wavelength, Read Mode, ROI
```

Reply:
I think he was referring to params being named in the match but not used?

Reply:
Ah right, I did a first version which used everything. I'm removing them.